### PR TITLE
Feature/web api wire mock

### PR DESCRIPTION
**Pull request by lstefaniszyn**
SOAP + REST   == Wiremock

Virtual api with example how to send/ recieve REST and SOAP messages

*How to review:*
1. Run Virtual server (Wiremock)
* DriverManager  https://github.com/devonfw/devonfw-testing/pull/102/files#diff-fce110ba3437d677d2a004b2a44d9f32
* DriverManagerTest https://github.com/devonfw/devonfw-testing/pull/102/files#diff-43514afadc2f01fbcf34eaab9f5bab91
* RuntimeParameters https://github.com/devonfw/devonfw-testing/pull/102/files#diff-ebc5a5ba7a9bd7b53605fb4a6a26a3ab

2. Create virtual response by mapping requests
* SOAP mapping https://github.com/devonfw/devonfw-testing/pull/102/files#diff-ff205638e8eb2616e0d79136989516a5
* SOAP test with mapped request with response https://github.com/devonfw/devonfw-testing/pull/102/files#diff-5a4e3f02a0695fa347f0a7d8306fb9dd
* REST mapping https://github.com/devonfw/devonfw-testing/pull/102/files#diff-9ab8acf8054ae7de7e15f6b08c3b0398
* REST test with mapped request with response https://github.com/devonfw/devonfw-testing/pull/102/files#diff-eca81f8ebbab33d19033108ef95f092d

3. SOAP e2e test. Mock response and send request 
* SOAPMessageGenerator  https://github.com/devonfw/devonfw-testing/pull/102/files#diff-b907dc70b7845ba940670d7ae0c7ec0b
* SOAP Test  https://github.com/devonfw/devonfw-testing/pull/102/files#diff-5a4e3f02a0695fa347f0a7d8306fb9dd
* SOAP Page files
FarenheitToCelsiusMethod_Response_FromFile.java - https://github.com/devonfw/devonfw-testing/pull/102/files#diff-a29c263717144adeb8dada96e7ea2bab
FarenheitToCelsiusMethod_Response_FromCode.java - https://github.com/devonfw/devonfw-testing/pull/102/files#diff-3a3bbbad99f3f14927ef8f78759a6c69
FarenheitToCelsiusMethod_Request_FromFile.java - https://github.com/devonfw/devonfw-testing/pull/102/files#diff-3564ebf8f8a3336919423ae75502b72e
FarenheitToCelsiusMethod_Request_FromCode.java - https://github.com/devonfw/devonfw-testing/pull/102/files#diff-a1357bec7d802226dc5dd80b43e84e7e

4. REST e2e test. Mock response and send request
* REST test https://github.com/devonfw/devonfw-testing/pull/102/files#diff-eca81f8ebbab33d19033108ef95f092d

### CODE REVIEW

File: allure-framework-modules/mrchecker-webapi-module/src/main/java/com/capgemini/ntc/endpoint/soap/FarenheitToCelsiusMethod_Request_FromFile.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+			throw new IOException("Could not find file. Path='" + path + "' does not exist");
+		}
+		
+		String responseEnvelope = readFile(path, StandardCharsets.UTF_8);
+		return responseEnvelope;
+	}
+	
+	private boolean exists(String path) {
+		File f = new File(path);
+		if (f.exists())
```
Comment (TSUPEL) [inline allure-framework-modules/mrchecker-webapi-module/src/main/java/com/capgemini/ntc/endpoint/soap/FarenheitToCelsiusMethod_Request_FromFile.java]: why not: 'return f.exists()' ?

File: allure-framework-modules/pom.xml [hunk 2]
```diff
@@ -43,6 +43,7 @@
         <module>allure-security-module</module>
         <module>mrchecker-webapi-module</module>
         <!-- <module>build-tools</module> -->
+        <module>mrchecker-webapi-module</module>
```
Comment (TSUPEL) [inline allure-framework-modules/pom.xml]: duplicate from line 44

File: allure-framework-modules/mrchecker-webapi-module/src/test/java/com/capgemini/ntc/webapi/core/base/driver/DriverManagerTest.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+			
+			// Enable to add new server instances in this thread. Simulation of multithread and bind to the same port
+			DriverManager.clearAllDrivers();
+			
+			// Start #2 server
+			driver2 = DriverManager.getDriverVirtualService();
+			assertTrue("Mock server does not run", driver2.isRunning());
+			assertEquals("Mock server for http does not run o port 8081", 8081, driver2.port());
+			assertTrue("Mock server for https does not run on random port", (Integer) driver2.httpsPort() instanceof Integer);
+			fail();
```
Comment (robert-duda-cap) [inline allure-framework-modules/mrchecker-webapi-module/src/test/java/com/capgemini/ntc/webapi/core/base/driver/DriverManagerTest.java]: Was it an intention to fail this test deliberately?

File: allure-framework-modules/mrchecker-webapi-module/src/test/java/com/capgemini/ntc/webapi/core/base/driver/DriverManagerTest.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+		} catch (FatalStartupException e) {
+			assertTrue("No information about bind error", e.getMessage()
+					.contains("Address already in use: bind"));
+		} finally {
+			// Close all drivers
+			try {
+				driver1.stop();
+				driver2.stop();
+			} catch (Exception e) {
```
Comment (robert-duda-cap) [inline allure-framework-modules/mrchecker-webapi-module/src/test/java/com/capgemini/ntc/webapi/core/base/driver/DriverManagerTest.java]: An empty catch.

File: allure-framework-modules/mrchecker-webapi-module/src/test/java/com/capgemini/ntc/webapi/core/base/properties/PropertiesFileSettingsTest.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+	public void setUp() throws Exception {
+	}
+	
+	@After
+	public void tearDown() throws Exception {
+		PropertiesSettingsModule.delInstance();
+	}
+	
+	@Test
+	public void testParamterEnableVirtualServer_True() {
```
Comment (robert-duda-cap) [inline allure-framework-modules/mrchecker-webapi-module/src/test/java/com/capgemini/ntc/webapi/core/base/properties/PropertiesFileSettingsTest.java]: Typo in word Parameter

File: allure-framework-modules/mrchecker-webapi-module/src/test/java/com/capgemini/ntc/webapi/core/base/properties/PropertiesFileSettingsTest.java [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+		String path = System.getProperty("user.dir") + Paths.get("/src/test/resources/settings.properties");
+		Injector i = Guice.createInjector(PropertiesSettingsModule.init(path));
+		this.propertiesFileSettings = i.getInstance(PropertiesFileSettings.class);
+		
+		assertEquals("", true, propertiesFileSettings.isVirtualServerEnabled());
+	}
+	
+	@Test
+	public void testParamterEnableVirtualServer_False() {
```
Comment (robert-duda-cap) [inline allure-framework-modules/mrchecker-webapi-module/src/test/java/com/capgemini/ntc/webapi/core/base/properties/PropertiesFileSettingsTest.java]: Typo in word Parameter

File: allure-framework-modules/mrchecker-webapi-module/src/test/java/com/capgemini/ntc/webapi/core/base/properties/PropertiesFileSettingsTest.java [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+		String path = System.getProperty("user.dir") + Paths.get("/src/test/resources/settings2.properties");
+		Injector i = Guice.createInjector(PropertiesSettingsModule.init(path));
+		this.propertiesFileSettings = i.getInstance(PropertiesFileSettings.class);
+		
+		assertEquals("", false, propertiesFileSettings.isVirtualServerEnabled());
+	}
+	
+	@Test
+	public void testParamterEnableVirtualServer_NoValue() {
```
Comment (robert-duda-cap) [inline allure-framework-modules/mrchecker-webapi-module/src/test/java/com/capgemini/ntc/webapi/core/base/properties/PropertiesFileSettingsTest.java]: Typo in word Parameter

File: allure-framework-modules/mrchecker-webapi-module/src/test/java/com/capgemini/ntc/webapi/core/base/properties/PropertiesFileSettingsTest.java [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+		String path = System.getProperty("user.dir") + Paths.get("/src/test/resources/settings3.properties");
+		Injector i = Guice.createInjector(PropertiesSettingsModule.init(path));
+		this.propertiesFileSettings = i.getInstance(PropertiesFileSettings.class);
+		
+		assertEquals("", true, propertiesFileSettings.isVirtualServerEnabled());
+	}
+	
+	@Test
+	public void testParamterEnableVirtualServer_Text() {
```
Comment (robert-duda-cap) [inline allure-framework-modules/mrchecker-webapi-module/src/test/java/com/capgemini/ntc/webapi/core/base/properties/PropertiesFileSettingsTest.java]: Typo in word Parameter

File: allure-framework-modules/mrchecker-webapi-module/src/test/java/com/capgemini/ntc/webapi/core/base/properties/PropertiesFileSettingsTest.java [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+		String path = System.getProperty("user.dir") + Paths.get("/src/test/resources/settings4.properties");
+		Injector i = Guice.createInjector(PropertiesSettingsModule.init(path));
+		this.propertiesFileSettings = i.getInstance(PropertiesFileSettings.class);
+		
+		assertEquals("", true, propertiesFileSettings.isVirtualServerEnabled());
+	}
+	
+	@Test
+	public void testParamterEnableVirtualServer_NoParameter() {
```
Comment (robert-duda-cap) [inline allure-framework-modules/mrchecker-webapi-module/src/test/java/com/capgemini/ntc/webapi/core/base/properties/PropertiesFileSettingsTest.java]: Typo in word Parameter

**Pull request merged**